aws/src/main/java/org/apache/iceberg/aws/s3/S3OutputFile.java

```diff
@@ -75,7 +95,7 @@ public PositionOutputStream createOrOverwrite() {
 
   @Override
   public InputFile toInputFile() {
-    return new S3InputFile(client(), uri(), null, s3FileIOProperties(), metrics());
+    return new S3InputFile(client(), asyncClient(), uri(), null, s3FileIOProperties(), metrics());
```

Comment:
This will use the new `asyncClient` by default. we should branch and gate this with `shouldUseAsyncClient`

Reply:
I think this is the async client passed to the S3 output file, which is already null if not enabling the accelerator?

Reply:
yep youre right. thanks